lms/djangoapps/grades/tests/test_api.py

```diff
@@ -108,3 +112,126 @@ def test_override_subsection_grade(self, earned_graded):
             else:
                 assert history_entry.history_user is None
                 assert history_entry.history_user_id is None
+
+
+class ClearGradeTests(ModuleStoreTestCase):
+    """
+    Tests for the clearing grades api call
+    """
+    @classmethod
+    def setUpTestData(cls):
+        super().setUpTestData()
+        cls.user = UserFactory()
+        cls.overriding_user = UserFactory()
+
+    @classmethod
+    def tearDownClass(cls):
+        super().tearDownClass()
+
+    def setUp(self):
+        super().setUp()
+        self.course = CourseFactory.create(org='edX', number='DemoX', display_name='Demo_Course', run='Spring2019')
+        self.subsection = BlockFactory.create(parent=self.course, category="sequential", display_name="Subsection")
+        self.grade = PersistentSubsectionGrade.update_or_create_grade(
+            user_id=self.user.id,
+            course_id=self.course.id,
+            usage_key=self.subsection.location,
+            first_attempted=None,
+            visible_blocks=[],
+            earned_all=6.0,
+            possible_all=6.0,
+            earned_graded=5.0,
+            possible_graded=5.0
+        )
+        self.params = {
+            "user_id": self.user.id,
+            "course_id": self.course.id,
+            "course_version": "JoeMcEwing",
+            "percent_grade": 77.7,
+            "letter_grade": "Great job",
+            "passed": True,
+        }
+        PersistentCourseGrade.update_or_create(**self.params)
+
+    def tearDown(self):
+        super().tearDown()
+        PersistentSubsectionGradeOverride.objects.all().delete()  # clear out all previous overrides
+
+    def test_clear_user_course_grades(self):
+        api.override_subsection_grade(
+            self.user.id,
+            self.course.id,
+            self.subsection.location,
+            overrider=self.overriding_user,
+            earned_graded=0.0,
+            comment='Test Override Comment',
+        )
+        override_obj = api.get_subsection_grade_override(
+            self.user.id,
+            self.course.id,
+            self.subsection.location
+        )
+        course_grade = PersistentCourseGrade.read(self.user.id, self.course.id)
+        self.assertIsNotNone(course_grade)
+        self.assertIsNotNone(override_obj)
+
+        api.clear_user_course_grades(self.user.id, self.course.id)
+
+        with self.assertRaises(PersistentCourseGrade.DoesNotExist):
+            PersistentCourseGrade.read(self.user.id, self.course.id)
+
+        with self.assertRaises(PersistentSubsectionGrade.DoesNotExist):
+            api.get_subsection_grade_override(
+                self.user.id,
+                self.course.id,
+                self.subsection.location
+            )
+
+    def test_clear_wrong_user_course_grades(self):
+        wrong_user = UserFactory()
```

Comment:
This test is functionally good, but I'd just add some comments walking through what specifically you're doing